src/src/SessionStore.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+** EPITECH PROJECT, 2022
+** SessionStore.cpp
+** File description:
+** SessionStore.cpp
+*/
+
+
+#include "Polymorph/Network/SessionStore.hpp"
+#include "authorizationKey.hpp"
+#include "Polymorph/Network/exceptions/UnknownAuthorizationKeyException.hpp"
+#include "Polymorph/Network/exceptions/UnauthorizedException.hpp"
+#include "Polymorph/Network/exceptions/AlreadyRegisteredException.hpp"
+#include "Polymorph/Network/exceptions/UnknownSessionException.hpp"
+
+void polymorph::network::SessionStore::registerAuthoredClient(asio::ip::udp::endpoint endpoint,
+        polymorph::network::SessionId sessionId, polymorph::network::AuthorizationKey key)
+{
+    try {
+        std::unique_lock<std::mutex> lock(_udpSessionsAuthorizationKeysMutex);
+        auto verifyKey = _udpSessionsAuthorizationKeys.at(sessionId);
```

Comment:
Using contains or any conditional presence will be better than an entire try catch block